graph/src/components/ethereum/types.rs

```diff
@@ -81,32 +81,61 @@ pub struct EthereumCall {
     pub transaction_hash: Option<H256>,
 }
 
-impl From<&Trace> for EthereumCall {
-    fn from(trace: &Trace) -> Self {
-        let (from, to, value, input) = match &trace.action {
-            Action::Call(call) => (call.from, call.to, call.value, call.input.clone()),
-            _ => (
-                Address::zero(),
-                Address::zero(),
-                U256::zero(),
-                Bytes::from(vec![]),
-            ),
+impl EthereumCall {
+    pub fn try_from_trace(trace: &Trace) -> Option<Self> {
+        // Filter out traces from operations which produced errors
```

Comment:
I'd still like to know more here, in which conditions is this error produced? Are they determinstic? Could this cause us to miss a call trigger?

Reply:
Nope, is error exists on the Trace it's because it was an operation that failed.